client/src/lib/utils.ts

```diff
@@ -1,6 +1,34 @@
+import {
+  WidgetData,
+  WidgetNavigationData,
+} from "@shared/dto/widgets/base-widget-data.interface";
 import { type ClassValue, clsx } from "clsx";
 import { twMerge } from "tailwind-merge";
 
 export function cn(...inputs: ClassValue[]) {
   return twMerge(clsx(inputs));
 }
+
+export const isWidgetData = (
+  data: WidgetData | WidgetNavigationData,
+): data is WidgetData => {
+  return (
+    Array.isArray(data) &&
+    (data.length === 0 ||
+      (typeof data[0] === "object" &&
+        "value" in data[0] &&
+        "total" in data[0] &&
+        "label" in data[0]))
+  );
+};
+
+export const isWidgetNavigationData = (
+  data: WidgetData | WidgetNavigationData,
+): data is WidgetNavigationData => {
+  return (
+    typeof data === "object" &&
+    data !== null &&
+    "href" in data &&
+    typeof data.href === "string"
+  );
+};
```

Comment:
I think we can avoid this by splitting properly the data input in every component. For example, if I have a `WidgetNavigation` component and any `data` I pass to it I consider it as `WidgetNavigationData` there's no need to finding out if the data can be anything else.

Anyways, in case of keeping this. I would move it to a more specific folder, like the `widgets` one. Putting it here feels too generic, IMO